File: go-leetcode/linked-list/141.环形链表.go
package leetcode

func hasCycle(head *ListNode) bool {
	visited := make(map[*ListNode]bool)

	curr := head

	for curr != nil {
		if visited[curr] {
			return true
		}
		visited[curr] = true
		curr = curr.Next
	}

	return false
}

File: python-leetcode/linked-list/141.环形链表.py
from typing import Optional

from type import ListNode


class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        visited = set()

        curr = head

        while curr:
            if curr in visited:
                return True
            visited.add(curr)
            curr = curr.next

        return False
